crates/ty_python_semantic/src/types/generics.rs

```diff
@@ -259,12 +290,13 @@ impl<'db> GenericContext<'db> {
         db: &'db dyn Db,
         typevar: TypeVarInstance<'db>,
     ) -> Option<TypeVarInstance<'db>> {
-        assert!(typevar.is_legacy(db));
+        assert!(typevar.is_legacy(db) || typevar.is_implicit(db));
         let typevar_def = typevar.definition(db);
         self.variables(db)
             .iter()
             .find(|self_typevar| {
-                self_typevar.is_legacy(db) && self_typevar.definition(db) == typevar_def
+                (self_typevar.is_legacy(db) || self_typevar.is_implicit(db))
+                    && self_typevar.definition(db) == typevar_def
```

Comment:
I must be missing something, but it's not clear to me how this works for implicit `Self` (when it's not annotated explicitly).